platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/GossipEvent.java

```diff
@@ -39,6 +42,13 @@ public class GossipEvent implements EventIntakeTask, BaseEvent, ChatterEvent {
     private Instant timeReceived;
     private long roundCreated = ROUND_CREATED_UNDEFINED;
 
+    /**
+     * The id of the node that sent this event to us.
+     * <p>
+     * Note: this is not necessarily the creator of this event: just the id of the node that this event was received from.
+     */
+    private NodeId senderNodeId;
```

Comment:
Here is an idea. What if instead of storing the sending node ID, we instead store a reference to the atomic integer that manages the count? The memory footprint is the exact same. If `null`, then the event was not received through gossip. 

Then, put the following methods into this class:

```
// Called when an event is received through gossip
public void setInakeCounter(AtomicLong intakeCounter) {
    this.intakeCounter = intakeCounter;
}
```

```
public void exitIntakePipeline() {
    if (intakeCounter != null) {
        intakeCounter.getAndDecrement();
    }
}
```

The benefit is that parts of the code that need to signal that the event has exited the pipeline no longer need a reference to the map of counters, they can simply discard an event by calling a method on it directly.

Reply:
This was a very good simplification - done